src/odb/src/db/dbInsertBuffer.h

```diff
@@ -0,0 +1,212 @@
+                             dbModule*& current_module,
+                             dbModITerm*& top_mod_iterm,
+                             dbModITerm* next_mod_iterm);
+  void performFinalConnections(dbITerm* load_pin,
+                               dbITerm* drvr_term,
+                               dbModITerm* top_mod_iterm);
+
+  //------------------------------------------------------------------
+  // Helper functions for insertBufferBeforeLoads
+  //------------------------------------------------------------------
+  dbModule* validateLoadPinsAndFindLCA(std::set<dbObject*>& load_pins,
+                                       std::set<dbNet*>& other_dbnets,
+                                       bool loads_on_diff_nets) const;
+  void createNewFlatAndHierNets(std::set<dbObject*>& load_pins);
+  void rewireBufferLoadPins(std::set<dbObject*>& load_pins);
+  void placeBufferAtLocation(dbInst* buffer_inst,
+                             const Point* loc,
+                             std::set<dbObject*>& load_pins);
+  void setBufferAttributes(dbInst* buffer_inst);
+
+ private:
+  //------------------------------------------------------------------
+  // Debug logging functions
+  //------------------------------------------------------------------
+  void dlogBeforeLoadsParams(const std::set<dbObject*>& load_pins,
+                             const Point* loc,
+                             bool loads_on_diff_nets) const;
+  void dlogTargetLoadPin(dbObject* load_obj) const;
+  void dlogDifferentDbNet(dbNet* net) const;
+  void dlogLCAModule(dbModule* target_module) const;
+  void dlogDumpNets(const std::set<dbNet*>& other_dbnets) const;
+  void dlogCreatingNewHierNet(const char* base_name) const;
+  void dlogConnectedBufferInputFlat() const;
+  void dlogConnectedBufferInputHier(dbModNet* orig_mod_net) const;
+  void dlogConnectedBufferInputViaHierConnect(dbObject* drvr_term) const;
+  void dlogConnectedBufferOutput() const;
+  void dlogCreatingHierConn(int load_idx, int num_loads, dbITerm* load) const;
+  void dlogMovedBTermLoad(int load_idx, int num_loads, dbBTerm* load) const;
+  void dlogPlacingBuffer(dbInst* buffer_inst, const Point& loc) const;
+  void dlogInsertBufferSuccess(dbInst* buffer_inst) const;
+  void dlogInsertBufferStart(int count, const char* mode) const;
+  void dlogSeparator() const;
+
+ private:
+  dbNet* net_{nullptr};  // Target net. The net to which the buffer is inserted
+  dbBlock* block_{nullptr};
+  utl::Logger* logger_{nullptr};
+
+  // Insert buffer state
+  dbITerm* buf_input_iterm_{nullptr};   // Input iterm of the new buffer
+  dbITerm* buf_output_iterm_{nullptr};  // Output iterm of the new buffer
+  dbNet* new_flat_net_{nullptr};    // New flat net connected to the new buffer
+  dbModNet* new_mod_net_{nullptr};  // New modnet connected to the new buffer
+  dbModule* target_module_{nullptr};        // Target module for the new buffer
+  const dbMaster* buffer_master_{nullptr};  // Buffer master
```

Comment:
Comments that don't add any information should be dropped (buffer_master_ is obviously the buffer master).